cmake/modules/AddSwift.cmake

```diff
@@ -623,8 +629,8 @@ function(_add_swift_library_single target name)
 
   # The section metadata objects are generated sources, and we need to tell CMake
   # not to expect to find them prior to their generation.
-  if("${CMAKE_SYSTEM_NAME}" STREQUAL "Linux" OR
-     "${CMAKE_SYSTEM_NAME}" STREQUAL "FreeBSD")
+  if("${SWIFTLIB_SINGLE_SDK}" STREQUAL "Linux" OR
+     "${SWIFTLIB_SINGLE_SDK}" STREQUAL "FreeBSD")
```

Comment:
`STREQUAL` is case-sensitive.  @rintaro is right, string constants should be uppercase.